.github/workflows/pr-tests.yml

```diff
@@ -0,0 +1,47 @@
+name: PR Tests
+
+on:
+  pull_request:
+    branches:    
+      - main
+
+jobs:
+  check-version:
+    name: Check version bumped
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Initialize the environment
+        uses: ./.github/actions/initialize
+      - name: Check version
+        run: python scripts/validate_version_bump.py
+  linting:
+    name: Run linting with flake8
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Initialize the environment
+        uses: ./.github/actions/initialize
+      - name: Lint the src/ directory
+        run: flake8 src/
+  check-deprecated:
+    name: Find code marked for removal in this version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Initialize the environment
+        uses: ./.github/actions/initialize
+      - name: Deprecated check
+        run: derp src/ src/citrine/__version__.py
+  run-tests:
+    name: Execute unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

Comment:
Is there no way to have a low-library-version / high-library-version dimension on the test matrix?  We're doing that now because it has caught problems with multiple libraries in the past.

Reply:
I presume you're talking about the `UPGRADE` thing?

As we discussed in Slack, I've yet to be shown any value it's given us. The issues you've pointed me to involve us relying on pre-1.0 libraries, and them breaking something on a minor version bump. Our setup.py should probably constrain them tighter to avoid those problems.

If we want to test for stuff like that, then we can set up some other build, but we shouldn't block PRs for unsupported library versions.